src/Sylius/Bundle/ResourceBundle/Form/DataTransformer/ResourceToIdentifierTransformer.php

```diff
@@ -49,7 +49,7 @@ public function __construct(RepositoryInterface $repository, $identifier = 'id')
     public function transform($value)
     {
         if (null === $value) {
-            return '';
+            return null;
```

Comment:
There are some exceptions even in Symfony codebase:
 - `BooleanToStringTransformer`
 - `ChoiceToValueTransformer`

I guess the convention was made to overcome this bug: https://github.com/symfony/symfony/pull/18357